### PR TITLE
Feature/authentication start

### PR DESCRIPTION
**Pull request by PaulKovalov**
I've recreated this PR as the old one had broken diff

### CODE REVIEW

File: portfolio/src/main/webapp/comments.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
 /* this file is a set of methods for fetching/creating comments */
 
+var authState = {};
```
Comment (harry-7) [inline portfolio/src/main/webapp/comments.js:3]: global variables are code smell. Generally, pass these global variables directly into methods that need it.

File: portfolio/src/main/webapp/comments.js [hunk 2]
```diff
@@ -143,12 +157,53 @@ function toggleReplyField(commentId) {
...
+
+// either adds log in button, or log out button. if user is
+// authenticated, displays user's email in DOM
+function addActionButtonsBasedOnAuthState() {
+  const authDiv = document.getElementById('auth');
+  if (authState.authenticated === true) {
+    const logOutLink = document.createElement('a');
+    const userEmailElement = document.createElement('p');
+    userEmailElement.innerText = 'Hello, ' + authState.email + ' !';
+    logOutLink.href = authState.logoutUrl;
```
Comment (harry-7) [inline portfolio/src/main/webapp/comments.js:178]: Suggestion: Prefer buttons for logout and login.

Comment (PaulKovalov) [reply portfolio/src/main/webapp/comments.js:178]: I used `a` because server sends a single link, and I don't have any logic related to it yet

Comment (PaulKovalov) [reply portfolio/src/main/webapp/comments.js:178]: Why do you want to use `button`?

Comment (harry-7) [reply portfolio/src/main/webapp/comments.js:178]: You can use button with a link :)

File: portfolio/src/main/webapp/comments.js [hunk 3]
```diff
@@ -16,28 +18,37 @@ async function fetchComments() {
...
+    // if there are no comments, add tag 'no comments'
+    createNoCommentsTag();
   }
+  // if client is authenticated, add new comment form
+  if (authState.authenticated) {
+    createNewCommentForm();
+    if (commentsJson.length) {
+      // if there are some comments, add button to delete all of them
+      // only authenticated user can delete comments (maybe will change later to administrator only feature)
+      addDeleteButton();
```
Comment (harry-7) [inline portfolio/src/main/webapp/comments.js:31]: If you are implementing delete based on auth, then isn't it better to provide functionality such that you can delete a specific comment and not all of your comments?

Comment (PaulKovalov) [reply portfolio/src/main/webapp/comments.js:31]: it is implemented in the further PR

**Pull request merged**